compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -132,6 +133,24 @@ struct ROCmOptions {
   }
 };
 
+// Extracts the amdgpu GfxIp version from the chip architecture in the
+// executable target attribute. For example, returns 0x942 for 'gfx942'.
+static FailureOr<unsigned> getGfxIpVersion(ExecutableTargetAttr targetAttr) {
+  IREE::GPU::TargetAttr gpuTarget = getGPUTargetAttr(targetAttr);
+  if (!gpuTarget)
+    return failure();
+
+  StringRef arch = gpuTarget.getArch();
+  if (!arch.consume_front("gfx"))
+    return failure();
+
+  unsigned gfxVersion = 0;
+  if (!llvm::to_integer(arch, gfxVersion, 16))
```

Comment:
1. I'm pretty sure this function is hanging out elsewhere in the code
2. The parsing isn't usually base-16. It's the rather annoying "last two digits are in hex, everything before that is decimal"

Reply:
(and we might want upstream's `amdgpu::Chipset` if we're feeling weirdly principled)

Reply:
Oh you mean this helper? https://github.com/llvm/llvm-project/blob/e04d124a96311d3714522125bf703950863e0540/mlir/include/mlir/Dialect/AMDGPU/Utils/Chipset.h

Thanks, I didn't know we have it already.